lib/accountregistry.h

```diff
@@ -0,0 +1,58 @@
+// SPDX-FileCopyrightText: 2020 Kitsune Ral <Kitsune-Ral@users.sf.net>
+// SPDX-FileCopyrightText: Tobias Fella <fella@posteo.de>
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#pragma once
+
+#include <QtCore/QObject>
+#include <QtCore/QList>
+#include <QtCore/QAbstractListModel>
+
+namespace Quotient {
+class Connection;
+
+class AccountRegistry : public QAbstractListModel {
+    Q_OBJECT
+public:
+    using const_iterator = QMap<QString, Connection *>::const_iterator;
+
+    enum EventRoles {
+        ConnectionRole = Qt::UserRole + 1,
+    };
+
+    static AccountRegistry &instance() {
+        static AccountRegistry _instance;
+        return _instance;
+    }
+
+    const QList<Connection*> accounts() const;
+    void add(Connection* a);
+    void drop(Connection* a);
+    bool isLoggedIn(const QString& userId) const;
+    Connection* operator[](int);
+    Connection* operator[](int) const;
+    bool isEmpty() const;
+    int count() const;
+    bool contains(Connection*) const;
+    Connection* get(const QString& userId);
+
+    const_iterator cbegin() const;
```

Comment:
This is very dangerous. See this fun thread https://mail.kde.org/pipermail/kde-pim/2021-April/047417.html

Tldr; this is not abi compatible if quotient is compiled with some flags but neochat/quatermion with another flag